packages/swarm-components/src/TextInput.jsx

```diff
@@ -50,6 +53,10 @@ type Props = {
    * Default size is `xs`
    */
   iconSize?: $Values<typeof ICON_SIZES>,
+  /**
+   * max length of input field
+   */
+  maxLength?: string | number,
```

Comment:
what does a `string` max length mean? I'd recommend locking down the type as much as possible, although if you need to support `string`, it might be worth a code comment about what that would look like (is it just numbers-as-strings? sounds awkward). Also, there's no unit test for a string-based `maxLength`, so that would be good to add if you need it

Reply:
This is taken from `Textarea`. I assume it's because some places in `mup-web` the maxLength is passed as a string, and in other places it's passed as a number. I agree it's awkward and should just be `number`. @hsbacot can you chime in on this please?

Reply:
`number` only makes sense. iirc I was having errors when integrating because it was passes as a string so this might reintroduce flow errors.

Reply:
okay i'm going to update the types for BOTH textarea and textinput then